app/(auth)/login/forgot-password/page.tsx

```diff
@@ -10,9 +10,14 @@ export default async function ForgotPassword({
 }) {
   const supabase = newServerClient();
 
-  let origin = headers().get('origin');
+  const headersList = await headers();
+  let origin: string | null = headersList.get('origin') || null;
+
   if (!origin) {
-    origin = 'http://localhost:3000/';
+    origin =
+      process.env.NODE_ENV === 'development'
+        ? 'http://localhost:3000'
+        : process.env.NEXT_PUBLIC_SITE_URL || null;
```

Comment:
I'm not entirely sure we need to have this `if` statement at all but definitely we don't want to evaluate to `process.env.NEXT_PUBLIC_SITE_URL` or `null` - we want to get rid of any possible `null`s here so we don't send people to a 404